src/set_theory/game/impartial.lean

```diff
@@ -0,0 +1,238 @@
+/-
+Copyright (c) 2020 Fox Thomson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Fox Thomson
+-/
+import set_theory.game.winner
+import tactic.nth_rewrite.default
+import tactic
+
+universe u
+
+/-!
+# Basic deinitions about impartial (pre-)games
+
+We will define an impartial game, one in which left and right can make exactly the same moves.
+Our definition differs slightly by saying that the game is always equivalent to its negative,
+no matter what moves are played. This allows for games such as poker-nim to be classifed as
+impartial.
+-/
+
+namespace pgame
+
+local infix ` ≈ ` := equiv
+
+/-- The definiton for a impartial game, defined using Conway induction -/
+def impartial : pgame → Prop
+| G := G ≈ -G ∧ (∀ i, impartial (G.move_left i)) ∧ (∀ j, impartial (G.move_right j))
+using_well_founded {dec_tac := pgame_wf_tac}
+
+lemma zero_impartial : impartial 0 := by tidy
+
+lemma impartial_def {G : pgame} :
+  G.impartial ↔ G ≈ -G ∧ (∀ i, impartial (G.move_left i)) ∧ (∀ j, impartial (G.move_right j)) :=
+begin
+  split,
+  { intro hi,
+    unfold1 impartial at hi,
+    exact hi },
+  { intro hi,
+    unfold1 impartial,
+    exact hi }
+end
+
+lemma impartial_neg_equiv_self {G : pgame} (h : G.impartial) : G ≈ -G := (impartial_def.1 h).1
+
+lemma impartial_move_left_impartial {G : pgame} (h : G.impartial) (i : G.left_moves) :
+  impartial (G.move_left i) :=
+(impartial_def.1 h).2.1 i
+
+lemma impartial_move_right_impartial {G : pgame} (h : G.impartial) (j : G.right_moves) :
+  impartial (G.move_right j) :=
+(impartial_def.1 h).2.2 j
+
+lemma impartial_add : ∀ {G H : pgame}, G.impartial → H.impartial → (G + H).impartial
+| G H :=
+begin
+  intros hG hH,
```

Comment:
Just do this in the pattern match? Also below.

Reply:
If I move this to the pattern match `pgame_wf_tac` doesn't work